plugins/module_utils/ilo_redfish_utils.py

```diff
@@ -51,183 +104,2123 @@ def get_ilo_sessions(self):
+                "msg": "Logical drives with these names were not found on the server: %s " % str(logical_drives_names_list)
+            }
+
+        result["logical_drives_details"] = needed_logical_drives
+
+        return {
+            "ret": True,
+            "msg": result
+        }
+
+    def get_nvme_drives_details(self):
+        # This method returns NVMe drive details
+        result = {}
+        nvme_drives_list = []
+        nvme_drive_url = self.root_uri + self.service_root + "systems/1/Storage/"
+        init_data = self.get_request(nvme_drive_url)
+        if not init_data["ret"]:
+            return init_data
+
+        if init_data["data"]["Members@odata.count"] == 0:
+            result["nvme_drives_count"] = 0
+            result["nvme_drives_list"] = []
+            return {
+                "ret": True,
+                "msg": result
+            }
+        else:
+            nvme_drives_count = init_data["data"]["Members@odata.count"]
+
+        # Get Members of Storage
+        for mem in init_data["data"]["Members"]:
+            mem_url = mem["@odata.id"]
+            mem_res = self.get_request(self.root_uri + mem_url)
+            if not mem_res["ret"]:
+                return mem_res
+
+            # Get Storage drives URI
+            if 'Drives' in mem_res['data']:
+                drives_url = mem_res['data']['Drives']
+            else:
+                return {
+                    "ret": False,
+                    "msg": "Physical drive URI not found in %s response %s" % (str(mem_url), str(mem_res))
+                }
+
+            # Get NVMe drive details
+            temp_data = mem_res
+            drives = []
+            for url in drives_url:
+                resp_NVMe = self.get_request(self.root_uri + url["@odata.id"])
+                if not resp_NVMe["ret"]:
+                    return resp_NVMe
+                drives.append(resp_NVMe)
+            temp_data["Drives"] = drives
+            nvme_drives_list.append(temp_data)
+
+        result["nvme_drives_count"] = nvme_drives_count
+        result["nvme_drives_list"] = nvme_drives_list
+
+        return {
+            "ret": True,
+            "msg": result
+        }
+
+    def validate_engine_id(self, user):
+        # Validating user Engine ID
+
+        error_msg = "Provided invalid engine ID: '%s'. " \
+                    "User Engine ID must be a hexadecimal string" \
+                    " with an even number of 10 to 64 characters, " \
+                    "excluding the first two characters, " \
+                    "0x (for example, 0x0123456789abcdef)"
+
+        if not user['user_engine_id']:
+            return {
+                "ret": False,
+                "msg": error_msg % (str(user['user_engine_id']))
+            }
+
+        engine_id = list(user['user_engine_id'])
+        if (len(engine_id) < 12 or len(engine_id) > 66) or \
+                ('0' not in engine_id[0] or engine_id[1] not in ['x', 'X']) or \
```

Comment:
better as:
```suggestion
                not engine_id[0].startswith("0x") or \
```

Reply:
Changed to not engine_id.startswith("0x")